ckanext/dia/commands.py

```diff
@@ -69,22 +57,55 @@ def cleanup_datastore_once(self, start_offset):
             sys.exit(1)
 
         # query datastore to get all resources from the _table_metadata
-        resource_id_list = []
-        total_tested = 0
-        for offset in itertools.count(start=start_offset, step=100):
-            print "Load metadata records from datastore (offset: %s)" % offset
-            record_list, tested = self._get_datastore_table_page(context, offset)  # noqa
-            total_tested += tested
-            resource_id_list.extend(record_list)
-            if not tested > 0:
-                break
-            if len(resource_id_list) > 250:
+        print("Loading metadata records from datastore")
+        all_datastore_tables = self._get_all_datastore_tables(context)
+
+        chunk_size = 100
+        total_deleted = 0
+        failed_deletes = []
+        for offset in itertools.count(start=0, step=chunk_size):
+            end_chunk_index = offset + chunk_size
+            datastore_tables_chunk = all_datastore_tables[offset:end_chunk_index]
+            orphan_list = self._find_orphaned_datastore_tables(context, datastore_tables_chunk)
+            if len(orphan_list) > 0:
                 # Run a small chunk of the dataset to avoid locking up the
                 # database for a *really* long time(read: until postgres is
                 # restarted)
+                delete_count, error_list = self._delete_orphans(context, orphan_list)
+                total_deleted += delete_count
+                failed_deletes.extend(error_list)
```

Comment:
Is a test needed here to see if error_list has something in it? Or does extend not matter for empty lists?

Reply:
error list is initialised as an empty array in the `_delete_orphans` and yep, in python it's totally fine to `[].extend([])`